File: greedy/javascript/scarecrow.js
/**
 * @param {number} n
 * @param {string} field
 * @return {number}
 */
var minimumScarecrows = function(n, field) {
    let ans = 0;
    for(let i = 0; i < n; i++){
        if(field[i] == '.'){
            i += 2;
            ans++;
        }
    }
    return ans;
}

console.log(minimumScarecrows(3, '.#.'))
console.log(minimumScarecrows(11, '...##....##'))
console.log(minimumScarecrows(2, '##'))

File: greedy/python/scarecrow.py
class Solution:
    def minimumScarecrows(self, n, field):
        ans = 0
        i = 0
        while i < n:
            if field[i] == '.':
                i += 3
                ans += 1
            else:
                i += 1
        return ans
    
s = Solution()
print(s.minimumScarecrows(3, '.#.'))
print(s.minimumScarecrows(11, '...##....##'))
print(s.minimumScarecrows(2, '##'))